### PR TITLE
feat: add C API to `@stdlib/math/base/special/lucas`

### LINKED ISSUE #892: [RFC]: Add C implementation for `@stdlib/math/base/special/lucas`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/lucas`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/lucas)

```c
double stdlib_base_lucas( const int32_t x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Jordan-Gallivan**
Resolves #892 .

## Description implementation of the C stdlib_base_lucas

> What is the purpose of this pull request?

This pull request: 

- implements all C files associated with the lucas function

## Related Issues

> Does this pull request have any related issues? No

This pull request:

-   resolves #892 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

@kgryte Wanted to give you a heads up that, when viewed in dark mode, the initial definition of the READMEs are hard to see (they're in black font).  

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/lucas/README.md [hunk 1]
```diff
@@ -135,6 +135,94 @@ for ( i = 0; i < 77; i++ ) {
...
+#include "stdlib/math/base/special/lucas.h"
+```
+
+#### stdlib_base_lucas( n )
+
+Computes the nth [Lucas number][lucas-number].
+
+```c
+double out = stdlib_base_lucas( 0 );
+// returns 0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/lucas/README.md]:
```suggestion
// returns 2
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Jordan-Gallivan!

**Comment by kgryte** [comment]
@Jordan-Gallivan Re: dark mode. Is this when viewing the Markdown on GitHub or elsewhere?

**Comment by kgryte** [comment]
You mind posting a screenshot?

**Comment by Jordan-Gallivan** [comment]
@kgryte 
See attached:
Screenshot 2023-02-28 at 10 47 36 AM

**Comment by kgryte** [comment]
Ah, thanks for sharing. Hmm...I am not sure what we can do about this, as the equations are in SVG. Maybe we can hook into GitHub's theming. Will add to our internal todo tracker. Thanks for flagging!

**Pull request merged**